File: Chapter 12/Dynamic Scale Beta-t-EGARCH/R/Info_Crit.R
# Information criteria used by the DCSt model.

Info_Crit <- function(Logl, par, n) {
  BIC <- -2 * Logl + length(par) * log(n)
  AIC <- 2 * length(par) - 2 * Logl
  list(BIC = BIC, AIC = AIC)
}

File: Chapter 12/Dynamic Scale Beta-t-EGARCH/Python/info_crit.py
"""Information criteria used by the DCSt model."""

import numpy as np


def info_crit(logl, par, n):
    """Return BIC and AIC using the likelihood convention supplied by the caller."""
    n_parameters = np.asarray(par).size
    bic = -2 * logl + n_parameters * np.log(n)
    aic = 2 * n_parameters - 2 * logl
    return bic, aic
